packages/core/src/types.ts

```diff
@@ -2219,3 +2241,53 @@ export type PersistedMachineState<TState extends AnyState> = Pick<
     };
   };
 };
+
+export interface InspectedActorObject {
+  actorRef: AnyActorRef; // local-only
+  sessionId: string;
+  parentId?: string; // Session ID
+  systemId?: string; // Session ID
+  events: ActorCommunicationEvent[]; // events where targetId === sessionId
+  definition?: string; // JSON-stringified machine definition or URL
+  createdAt: number; // Timestamp
+  updatedAt: number; // Timestamp, derived from latest update createdAt
+  status: 0 | 1 | 2; // 0 = not started, 1 = started, 2 = stopped, derived from latest update status
+}
+
+export interface BaseInspectionEvent {
+  actorSystemId: string; // the session ID of the root
```

Comment:
This one would sound a little bit more accurate. `actorSystemId` sounds a little bit like if we'd have special system ids but we don't - the system is attached to the root actor but it has no id of its own.
```suggestion
  rootId: string; // the session ID of the root
```